orthopairs/src/main/java/org/reactome/release/orthopairs/OrthopairFileGenerator.java

```diff
@@ -0,0 +1,93 @@
+package org.reactome.release.orthopairs;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.simple.JSONObject;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.*;
+
+public class OrthopairFileGenerator {
+    private static final Logger logger = LogManager.getLogger();
+    // Create source-target protein mapping file
+    public static void createProteinHomologyFile(String speciesKey, String sourceTargetProteinMappingFilename, JSONObject speciesJSON, Map<String,Set<String>> speciesProteinHomologs ) throws IOException {
+
+        logger.info("\tGenerating " + sourceTargetProteinMappingFilename);
+        File sourceTargetProteinMappingFile = new File(sourceTargetProteinMappingFilename);
+        if (sourceTargetProteinMappingFile.exists()) {
+            sourceTargetProteinMappingFile.delete();
+        }
+        sourceTargetProteinMappingFile.createNewFile();
+
+        List<String> sourceProteinIds = new ArrayList<>();
+        sourceProteinIds.addAll(speciesProteinHomologs.keySet());
+        Collections.sort(sourceProteinIds);
+        for (String sourceProteinId : sourceProteinIds) {
+            Set<String> targetProteinIds = speciesProteinHomologs.get(sourceProteinId);
+            targetProteinIds.remove("LDO");
+            String proteinOrthologLine = sourceProteinId.split("=")[1] + "\t";
+            List<String> cleanTargetProteinIds = new ArrayList<>();
+            for (String targetProteinId : targetProteinIds) {
+                cleanTargetProteinIds.add(targetProteinId.split("=")[1]);
+            }
+            Collections.sort(cleanTargetProteinIds);
+            proteinOrthologLine += String.join(" ", cleanTargetProteinIds) + "\n";
+            Files.write(Paths.get(sourceTargetProteinMappingFile.getPath()), proteinOrthologLine.getBytes(), StandardOpenOption.APPEND);
+        }
+    }
+
+    // Create target species gene-protein mapping file
+    public static void createSpeciesGeneProteinFile(String speciesKey, String targetGeneProteinMappingFilename, JSONObject speciesJSON, Map<String,Set<String>> speciesGeneProteinMap) throws IOException {
+
+        logger.info("\tGenerating " + targetGeneProteinMappingFilename);
+        File targetGeneProteinMappingFile = new File (targetGeneProteinMappingFilename);
+        if (targetGeneProteinMappingFile.exists()) {
+            targetGeneProteinMappingFile.delete();
+        }
+
+        boolean altIdMappingExists = false;
```

Comment:
A method of the same name to query might be preferable to a boolean temp so there's no need to track the variable state.

```
private static boolean altIdMappingExists(JSONObject speciesJSON) {
	return speciesJSON.get("alt_id_file") != null;
}
```